server/src/internalClusterTest/java/org/elasticsearch/index/engine/ThreadPoolMergeSchedulerStressTestIT.java

```diff
@@ -82,7 +82,9 @@ public static class TestEnginePlugin extends Plugin implements EnginePlugin {
         void allowAllMerging() {
             // even when indexing is done, queued and backlogged merges can themselves trigger further merging
             // don't let this test be bothered by that, and simply let all merging run unhindered
-            runMergeSemaphore.release(Integer.MAX_VALUE - initialRunMergesCount);
```

Comment:
The semaphore was also released when excess merges were enqueued. This is a dynamic process. I preferred I not fudge an upper limit value, or put Integer.MAX_VALUE / 2 ...